pkg/pmaperrors/pmaperrors.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2023 PAMP authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package pmaperrors defines the sentinel errors for the project.
+package pmaperrors
```

Comment:
This file looks confusing to me. Can we define retryable error and non-retryable error similar to Can we use the definition similar to https://github.com/sethvargo/go-retry/blob/a99f7cdbe5d1697463eccaf888be309b9e8d1b5f/retry.go#L24?

Reply:
Updated.